### DISCUSSION TITLE
Getting aggregate counts of array elements across rows

### DISCUSSION THREAD

**Discussion by jschuur** [prisma/prisma#10869] [Q&A] [answered]
I'd like to count the total number of occurrences of each unique strings from a `String[]` type across multiple rows and do it all as part of a query. Can I do this using Prisma's current aggregate/groupBy functionality?

I'm not sure it can be done, but even if it's not, I also don't know how I would structure a query (*) to use in `$queryRaw`. FWIW, I'm using PostgreSQL. 

In my `schema.prisma` ([full file](https://github.com/jschuur/streamers.dev/blob/main/prisma/schema.prisma)), `allTags` is set up as an array of strings:

```
  allTags                 String[]
```

This then got turned into a _text in the database schema, and the underscore seems to mean it's an array there:

```
    "allTags" _text,
```

So assuming data like this:

```
allTags
------------------------------
{JavaScript, Vue.js}
{JavaScript, Node}
{TypeScript, JavaScript, Node}
```

...the desired results should look something like this:

```
count + tag
------------------
3     | JavaScript
2     | Node
1     | Vue.js
1     | TypeScript
```

\* I've [posted](https://stackoverflow.com/questions/70480275/postgresql-query-for-an-aggregate-count-of-distinct-elements-within-an-array) this to StackOverflow too, to get help on how this might be done in raw SQL.

**Comment by jschuur** [comment, marked as answer]
Over on Stack Overflow, I've gotten a raw query that solves my immediate question:

```
SELECT
    COUNT(*) AS "count", tag
FROM 
    "Stream" AS s
CROSS JOIN LATERAL UNNEST(s."allTags") AS tags(tag)
GROUP BY
    tag
ORDER BY 1 DESC;
```

If this can't be done in Prisma directly, I'm happy to use a `$queryRaw` and use this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
